### PR TITLE
Expose create_sparsitypattern

### PR DESCRIPTION
**Pull request by jorgensd**
Exposing create sparsity pattern to be able to manually append to pattern for multipointconstraints

### CODE REVIEW

File: cpp/dolfinx/fem/utils.cpp [hunk 1]
```diff
@@ -145,9 +144,21 @@ la::PETScMatrix dolfinx::fem::create_matrix(const Form& a)
   if (a.integrals().num_integrals(fem::FormIntegrals::Type::exterior_facet) > 0)
     SparsityPatternBuilder::exterior_facets(pattern, mesh,
```
Comment (garth-wells) [inline cpp/dolfinx/fem/utils.cpp]: Put curly braces around the multi-line statement.

File: cpp/dolfinx/fem/utils.h [hunk 2]
```diff
@@ -63,6 +63,9 @@ block_function_spaces(
 /// Create matrix. Matrix is not zeroed.
 la::PETScMatrix create_matrix(const Form& a);
 
+/// Create sparsity pattern for a given form. Does not assemble the pattern.
```
Comment (garth-wells) [inline cpp/dolfinx/fem/utils.h]: Add doxygen doc

File: python/dolfinx/wrappers/fem.cpp [hunk 3]
```diff
@@ -143,6 +144,15 @@ void fem(py::module& m)
       },
       py::return_value_policy::take_ownership,
       "Create nested vector for multiple (stacked) linear forms.");
+  m.def(
```
Comment (garth-wells) [inline python/dolfinx/wrappers/fem.cpp]: Does this need to be a lambda?

File: cpp/dolfinx/fem/utils.cpp [hunk 4]
```diff
@@ -155,6 +155,18 @@ la::PETScMatrix dolfinx::fem::create_matrix(const Form& a)
   pattern.assemble();
```
Comment (garth-wells) [inline cpp/dolfinx/fem/utils.cpp]: I think this line should be removed and let the caller finalise the sparsity pattern.

File: cpp/dolfinx/fem/utils.h [hunk 5]
```diff
@@ -52,17 +52,27 @@ class Form;
...
     const Eigen::Ref<const Eigen::Array<const fem::Form*, Eigen::Dynamic,
                                         Eigen::Dynamic, Eigen::RowMajor>>& a);
 
-/// Create matrix. Matrix is not zeroed.
+/// Create a matrix.
+/// @param[in] a  A bilinear form
+/// @return A matrix. The matrix is not zeroed.
 la::PETScMatrix create_matrix(const Form& a);
 
+/// Create a sparsity pattern for a given form. The pattern is no
```
Comment (jorgensd) [inline cpp/dolfinx/fem/utils.h]: @garth-wells Typo. Should be "The pattern is not"

**Pull request merged**